controller/internal/enforcer/nfqdatapath/datapath_udp.go

```diff
@@ -638,3 +659,48 @@ func (d *Datapath) processNetworkUDPAckPacket(udpPacket *packet.Packet, context
 
 	return nil, nil, nil
 }
+
+func (d *Datapath) checkForApplicationACLs(p *packet.Packet) (action *policy.FlowPolicy, err error) {
+
+	context, err := d.contextFromIP(true, p.DestinationAddress.String(), p.Mark, p.DestinationPort, packet.IPProtocolUDP)
+	if err != nil {
+		return nil, err
+	}
+
+	// check for network ACLS response
+	_, action, err = context.NetworkUDPACLPolicyFromAddr(p.DestinationAddress.To4(), p.SourcePort)
+	if err == nil {
+		return action, err
+	}
+
+	report, action, err := context.ApplicationUDPACLPolicyFromAddr(p.DestinationAddress.To4(), p.DestinationPort)
+	if err != nil {
+		return nil, err
+	}
+
+	d.reportExternalServiceFlow(context, report, action, true, p)
+	return action, nil
+}
+
+func (d *Datapath) checkForExternalServices(p *packet.Packet) (action *policy.FlowPolicy, err error) {
+
+	// check for app acl response - like a dns response
+	context, err := d.contextFromIP(false, p.DestinationAddress.String(), p.Mark, p.DestinationPort, packet.IPProtocolUDP)
+	if err != nil {
+		return nil, err
+	}
+
+	_, action, err = context.ApplicationUDPACLPolicyFromAddr(p.SourceAddress.To4(), p.SourcePort)
+	if err == nil {
+		return action, nil
+	}
+
+	// check for network ACLS
+	report, action, err := context.NetworkUDPACLPolicyFromAddr(p.SourceAddress.To4(), p.DestinationPort)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
Don't we know the direction? I can see this creating problems if I use 0.0.0.0. Why check for both directions?